src/access.rs

```diff
@@ -0,0 +1,320 @@
+use crate::error::LdkLiteError as Error;
+
+use lightning::chain::chaininterface::{BroadcasterInterface, ConfirmationTarget, FeeEstimator};
+use lightning::chain::WatchedOutput;
+use lightning::chain::{Confirm, Filter};
+
+use bdk::blockchain::{Blockchain, EsploraBlockchain, GetBlockHash, GetHeight, GetTx};
+use bdk::database::BatchDatabase;
+use bdk::wallet::AddressIndex;
+use bdk::{SignOptions, SyncOptions};
+
+use bitcoin::{BlockHash, Script, Transaction, Txid};
+
+use std::sync::Mutex;
+
+pub struct LdkLiteChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	blockchain: Mutex<EsploraBlockchain>,
+	wallet: Mutex<bdk::Wallet<D>>,
+	queued_transactions: Mutex<Vec<Txid>>,
+	watched_transactions: Mutex<Vec<Txid>>,
+	queued_outputs: Mutex<Vec<WatchedOutput>>,
+	watched_outputs: Mutex<Vec<WatchedOutput>>,
+	last_sync_height: Mutex<Option<u32>>,
+}
+
+impl<D> LdkLiteChainAccess<D>
+where
+	D: BatchDatabase,
+{
+	pub fn new(blockchain: EsploraBlockchain, wallet: bdk::Wallet<D>) -> Self {
+		let blockchain = Mutex::new(blockchain);
+		let wallet = Mutex::new(wallet);
```

Comment:
Hm, to be honest I'd rather not because in this initial step the individual components are pretty interdependent and due to change. I imagine it to be more annoying to constantly having to rebase the individual PRs on each other. 

So, if you not really insist, I'd like to keep it in this one PR for now and handle forthcoming changes in individual PRs.